drools-wb-services/drools-wb-verifier/drools-wb-verifier-guided-decision-table-adapter/src/test/java/org/drools/workbench/services/verifier/plugin/client/DecisionTableAnalyzerUpdateTest.java

```diff
@@ -86,6 +88,62 @@ public void testRowValueChange() throws Exception {
         assertTrue(analyzerProvider.getAnalysisReport().isEmpty());
     }
 
+    @Test
+    public void testUpdateDateEffective() throws Exception {
+
+        table52 = analyzerProvider.makeAnalyser()
+                .withAttributeColumn("date-effective")
+                .withAttributeColumn("date-expires")
+                .withPersonAgeColumn("==")
+                .withPersonApprovedActionSetField()
+                .withData(DataBuilderProvider
+                                  .row(new Date(2000, 11, 01), new Date(3000, 11, 02), 1, true)
+                                  .row(new Date(2000, 11, 03), new Date(5000, 11, 04), 1, true)
+                                  .end())
+                .buildTable();
+
+        fireUpAnalyzer();
+
+        assertContains(analyzerProvider.getAnalysisReport(),
+                       CheckType.REDUNDANT_ROWS,
+                       Severity.WARNING,
+                       1, 2);
+
+        setValue(1,
+                 2,
+                 new Date(4000, 11, 05));
+
+        assertTrue(analyzerProvider.getAnalysisReport().isEmpty());
+    }
+
+    @Test
+    public void testUpdateDateExpires() throws Exception {
+
+        table52 = analyzerProvider.makeAnalyser()
+                .withAttributeColumn("date-effective")
+                .withAttributeColumn("date-expires")
+                .withPersonAgeColumn("==")
+                .withPersonApprovedActionSetField()
+                .withData(DataBuilderProvider
+                                  .row(new Date(2000, 11, 01), new Date(3000, 11, 02), 1, true)
+                                  .row(new Date(1000, 11, 03), new Date(5000, 11, 04), 1, true)
+                                  .end())
+                .buildTable();
+
+        fireUpAnalyzer();
+
+        assertContains(analyzerProvider.getAnalysisReport(),
+                       CheckType.REDUNDANT_ROWS,
+                       Severity.WARNING,
+                       1, 2);
+
+        setValue(1,
```

Comment:
Shouldn't we update the second row to have two valid intervals?

Reply:
The index numbers might be misleading, but this change results in both ranges being valid (date-effective is before date-expires) and the ranges do not overlap. So Date with 5000 becomes 1500.